src/main/java/com/jvm_bloggers/admin_panel/moderation/FetchingDataPanel.java

```diff
@@ -58,9 +44,8 @@ private AjaxLink createFetchBlogPostsDataButton(CustomFeedbackPanel feedback) {
             @Override
             public void onClick(AjaxRequestTarget target) {
                 log.debug("Fetching blog posts data clicked");
-                if (fetchingBlogPostExecutor.getActiveCount() == 0) {
-                    fetchingBlogPostExecutor
-                        .submit(blogPostsFetchingScheduler::checkRssForNewBlogPosts);
+                if (!blogPostsFetcher.isFetchingProcessInProgress()) {
```

Comment:
Maybe `refreshPost` should return some kind of status marking request as `ACCEPTED` or `REJECTED`. Maybe even with appropriate message included.